src/main/java/org/traccar/WebDataHandler.java

```diff
@@ -155,32 +181,95 @@ public String formatRequest(Position position) throws UnsupportedEncodingExcepti
     @Override
     protected Position handlePosition(Position position) {
 
-        String url;
-        if (json) {
-            url = this.url;
-        } else {
-            try {
-                url = formatRequest(position);
-            } catch (UnsupportedEncodingException | JsonProcessingException e) {
-                throw new RuntimeException("Forwarding formatting error", e);
+        class AsyncRequestAndCallback implements InvocationCallback<Response> {
+
+            private int delay = retryDelayMin;
+            private Map<String, Object> payload;
+            private Invocation.Builder requestBuilder;
+
+            AsyncRequestAndCallback(Position position) {
+
+                String formattedUrl;
+                try {
+                    formattedUrl = json ? url : formatRequest(position);
+                } catch (UnsupportedEncodingException | JsonProcessingException e) {
+                    throw new RuntimeException("Forwarding formatting error", e);
+                }
+
+                requestBuilder = client.target(formattedUrl).request();
+                if (header != null && !header.isEmpty()) {
+                    for (String line: header.split("\\r?\\n")) {
+                        String[] values = line.split(":", 2);
+                        requestBuilder.header(values[0].trim(), values[1].trim());
+                    }
+                }
+
+                if (json) {
+                    payload = prepareJsonPayload(position);
+                }
+
+                deliveryPendingCurrent.incrementAndGet();
+
+                send();
             }
-        }
 
-        Invocation.Builder requestBuilder = client.target(url).request();
+            private void send() {
+                if (json) {
+                    requestBuilder.async().post(Entity.json(payload), this);
+                } else {
+                    requestBuilder.async().get(this);
+                }
+            }
 
-        if (header != null && !header.isEmpty()) {
-            for (String line: header.split("\\r?\\n")) {
-                String[] values = line.split(":", 2);
-                requestBuilder.header(values[0].trim(), values[1].trim());
+            private void retry() {
+                try {
+                    String message = "Position forwarding failed.";
+                    if (!retryEnabled) {
+                        LOGGER.warn(message);
+                    } else {
+                        int pending = deliveryPendingCurrent.get();
+                        if (pending <= deliveryPendingLimit) {
+                            LOGGER.warn(message + " Pending: " + pending
+                                + ". Retrying in " + delay + " seconds.");
+                            GlobalTimer.getTimer().newTimeout(new TimerTask() {
```

Comment:
Please use injection.

Reply:
Sorry, can you please be more specific?

Reply:
Don't use `GlobalTimer.getTimer()`. Use injection for any dependencies.

Reply:
Oh, I see..., I think I got it. Sorry, coming from C++